core/description/resource.go

```diff
@@ -15,7 +15,7 @@ type Resource interface {
 	Name() string
 	Revision() int
 	CharmStoreRevision() int
-	AddRevision(ResourceRevisionArgs) ResourceRevision
+	AddRevision(ResourceRevisionArgs) (ResourceRevision, error)
```

Comment:
would love to see comments on these at some point, especially to explain that AddRevision can accept a rev that already exists in which case it will be merged etc